### PR TITLE
fix: use relative uris in the client id and refer to the local client id

### PR DESCRIPTION
**Pull request by jeswr**
In theory this fixes https://github.com/comunica/comunica-feature-link-traversal/issues/100 but I cannot test without deploying to a URL since servers can't read clientIds from localhost.

Do you have an easy way of testing the website at a development url?

### CODE REVIEW

File: src/ldf-client-ui.js [hunk 1]
```diff
@@ -350,11 +351,12 @@ if (typeof global.process === 'undefined')
...
                 }
                 else {
+                  const href = window.location.href.replace('#', '?');
                   $solidSession.login({
                     oidcIssuer: $idp.val(),
-                    redirectUrl: window.location.href.replace('#', '?'), // OIDC does not allow hash fragments, so we encode it as query param
+                    redirectUrl: href, // OIDC does not allow hash fragments, so we encode it as query param
                     clientName: 'Comunica Web Client',
-                    clientId: 'https://query.linkeddatafragments.org/solid-client-id.json',
+                    clientId: join(href, 'solid-client-id.json'),
```
Comment (rubensworks) [inline src/ldf-client-ui.js:359]: We may not need this new dependency, it may be achievable using `relative-to-absolute-iri`

**Comment by rubensworks** [comment]
> Do you have an easy way of testing the website at a development url?

You could try setting it up at localhost, and overriding your `/etc/hosts` file to make a domain refer to localhost. (not 100% sure this will work)

**Comment by jeswr** [comment]
> and overriding your /etc/hosts file to make a domain refer to localhost. (not 100% sure this will work)

This won't work because it the Solid Server that needs to retrieve the client-id as part of the authentication flow.

**Comment by jeswr** [comment]
@rubensworks - I've now fixed this and been able to test in the localhost. It turns out the the clientID *must* use absolute URIs to comply with the OIDC spec so I've now got templating in place to create absolute URLs for the client ID document, and use those same URLs in the JavaScript login call.

The `baseURL` can now be overridden by the `-b` parameter so once this is merged all instances of the Comunica Website will need to be updated to provide that parameter (except for the one hosted at query.linkeddatafragments.org which is the default value for that parameter).

**Comment by rubensworks** [comment]
Released as 1.8.0.

**Pull request merged**